include/kernel.h

```diff
@@ -1795,6 +1989,17 @@ static inline void *z_impl_k_timer_user_data_get(struct k_timer *timer)
  * @{
  */
 
+/**
+ * @brief Get system uptime, in ticks.
+ *
+ * This routine returns the elapsed time since the system booted, in
+ * ticks (c.f. :option:`CONFIG_SYS_CLOCK_TICKS_PER_SEC`), which is the
+ * fundamental unit of resolution of kernel timekeeping.
+ *
+ * @return Current uptime in ticks.
+ */
+__syscall s64_t k_uptime_ticks(void);
```

Comment:
why not `u64_t`?

Reply:
To conform with k_uptime_get() mostly, which was (probably -- it predates me, but I agree) done to prevent the application author from computing a time delta in unsigned units and tripping over the resulting overflow.